From 3122a6046e5679d1b71367596f74a8d3d6e999d4 Mon Sep 17 00:00:00 2001
From: amistry <amistry@chromium.org>
Date: Tue, 9 Jun 2015 12:18:39 -0700
Subject: [PATCH 51/58] hotword: disable at build time by default

Commit f269d3b548203e217e8c0080c2e22e7ae3efb51e upstream:

"""
Hotwording downloads a shared module from the web store containing a
NaCl module. There is a desire to build and distribute Chromium
without this happening. This change adds an "enable_hotwording" build
flag that is enabled by default, but can be disabled at compile time.

BUG=491435
Review URL: https://codereview.chromium.org/1160243004
Cr-Commit-Position: refs/heads/master@{#333548}
"""

Hard-disable hotwording by default even on a build level.
---
 chrome/browser/BUILD.gn                           | 6 ++----
 chrome/browser/extensions/component_loader.cc     | 4 ++++
 chrome/browser/search/hotword_service.cc          | 1 -
 chrome/browser/search/hotword_service_unittest.cc | 4 ++++
 4 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 8b7e0fe..5345ca2 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -19,10 +19,8 @@ if (is_desktop_linux) {
 }
 
 declare_args() {
-  # 'Ok Google' hotwording is disabled by default in open source builds. Set to
-  # true to enable. (This will download a closed-source NaCl module at startup.)
-  # Chrome-branded builds have this enabled by default.
-  enable_hotwording = is_chrome_branded
+  # 'Ok Google' hotwording is NOT enabled.
+  enable_hotwording = false
 }
 
 about_credits_file = "$target_gen_dir/about_credits.html"
diff --git a/chrome/browser/extensions/component_loader.cc b/chrome/browser/extensions/component_loader.cc
index 029ad1e..0ff6842 100644
--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -347,17 +347,21 @@ void ComponentLoader::AddHangoutServicesExtension() {
 }
 
 void ComponentLoader::AddHotwordAudioVerificationApp() {
+#if 0
   if (HotwordServiceFactory::IsAlwaysOnAvailable()) {
     Add(IDR_HOTWORD_AUDIO_VERIFICATION_MANIFEST,
         base::FilePath(FILE_PATH_LITERAL("hotword_audio_verification")));
   }
+#endif
 }
 
 void ComponentLoader::AddHotwordHelperExtension() {
+#if 0
   if (HotwordServiceFactory::IsHotwordAllowed(browser_context_)) {
     Add(IDR_HOTWORD_MANIFEST,
         base::FilePath(FILE_PATH_LITERAL("hotword")));
   }
+#endif
 }
 
 void ComponentLoader::AddImageLoaderExtension() {
diff --git a/chrome/browser/search/hotword_service.cc b/chrome/browser/search/hotword_service.cc
index e93789b..b5a056a 100644
--- a/chrome/browser/search/hotword_service.cc
+++ b/chrome/browser/search/hotword_service.cc
@@ -649,7 +649,6 @@ bool HotwordService::IsHotwordAllowed() {
   // set.
   if (group == hotword_internal::kHotwordFieldTrialDisabledGroupName)
     return false;
-
   return DoesHotwordSupportLanguage(profile_);
 #else
   return false;
diff --git a/chrome/browser/search/hotword_service_unittest.cc b/chrome/browser/search/hotword_service_unittest.cc
index 521bb7b..585cfab 100644
--- a/chrome/browser/search/hotword_service_unittest.cc
+++ b/chrome/browser/search/hotword_service_unittest.cc
@@ -219,6 +219,7 @@ TEST_P(HotwordServiceTest, DISABLED_IsHotwordAllowedInvalidFieldTrial) {
 
 // Disabled due to http://crbug.com/503963.
 TEST_P(HotwordServiceTest, DISABLED_IsHotwordAllowedLocale) {
+#if defined(ENABLE_HOTWORDING)
   TestingProfile::Builder profile_builder;
   scoped_ptr<TestingProfile> profile = profile_builder.Build();
 
@@ -260,6 +261,7 @@ TEST_P(HotwordServiceTest, DISABLED_IsHotwordAllowedLocale) {
   Profile* otr_profile = profile->GetOffTheRecordProfile();
   SetApplicationLocale(otr_profile, "en");
   EXPECT_FALSE(HotwordServiceFactory::IsHotwordAllowed(otr_profile));
+#endif  // defined(ENABLE_HOTWORDING)
 }
 
 TEST_P(HotwordServiceTest, ShouldReinstallExtension) {
@@ -316,6 +318,7 @@ TEST_P(HotwordServiceTest, PreviousLanguageSetOnInstall) {
 }
 
 TEST_P(HotwordServiceTest, UninstallReinstallTriggeredCorrectly) {
+#if defined(ENABLE_HOTWORDING)
   InitializeEmptyExtensionService();
   service_->Init();
 
@@ -396,6 +399,7 @@ TEST_P(HotwordServiceTest, UninstallReinstallTriggeredCorrectly) {
 #endif
   EXPECT_FALSE(hotword_service->MaybeReinstallHotwordExtension());
   EXPECT_EQ(1, hotword_service->uninstall_count());  // no change
+#endif  // defined(ENABLE_HOTWORDING)
 }
 
 TEST_P(HotwordServiceTest, DisableAlwaysOnOnLanguageChange) {
-- 
2.4.3

